lib/internal/loader/DefaultResolve.js

```diff
@@ -55,9 +67,18 @@ function resolve(specifier, parentURL) {
     };
   }
 
-  let url;
+  const mainModeESM = parentURL === undefined &&
```

Comment:
@guybedford could we consider just checking the existence of "esm" as the field name. 

This makes things really akward in light of: https://github.com/defunctzombie/package-browser-field-spec and limits package authors the ability to ship multiple types that bundlers, and tools can leverage as fallbacks. 

For example: we fall back to other fields, if "esm" wouldn't be available etc, and then we can enforce strict ESM mode also.

Reply:
this check is done for the main entry point via the node CLI, it is not on the `"main"` field in `package.json` files.

Reply:
Correct, but I'd "mode" in the package.json is a sigil for "expect explitily this package as esm, then I'm asking to consider just "esm" as the name in the package not mode. And instead of a Boolean value to instead have "esm" point to the module entrypoint. This allows fallback for non esm modules in a single package.

Reply:
The original work at https://github.com/nodejs/node-eps/pull/60 exactly did look at implementing the "module main" principles, but there were a couple of issues with this:
1. Subpaths get confusing - should pkg/x.js, where "x.js" is not the main, be treated as a module or not? If not, how to specify subpaths, and if so, effectively one "module" property then changes the way every module in the package is interpreted, which may be surprising to some that it would break loading any CommonJS files in the package.
2. Interactions with other "main" fields get confusing - we found ourselves wanting to spec a "browserModule" main here etc etc... it quickly gets ridiculous.
 
So the next idea was simply to do "esm": true as boolean flag, but the issue again is that that may be too restrictive for future changes to the interpretations, so "mode": "esm" feels like an extensible safe way forward encapsulating the entire ruleset.

Reply:
@TheLarkInn is this still a requested change?